open_seq2seq/utils/funcs.py

```diff
@@ -118,7 +117,7 @@ def train(train_model, eval_model=None, debug_port=None):
                "train model does not define get_num_objects_per_step method.")
 
   # starting training
-  if fine_tuning:
+  if load_model_dir or tf.train.latest_checkpoint(checkpoint_dir):
```

Comment:
I only wanted to change the behaviour in case the "load_model" is given and nothing else, so I only changed (not load_model_dir) to load_model_dir. 

I assume that if there is a checkpoint in the specified folder (tf.train.latest_checkpoint(checkpoint_dir)) then we should be in the"continue_learning" mode, and hence the "TransferMonitoredTrainingSession" instead of the tf.train.MonitoredTrainingSession. Though it's unclear to me what the difference between these Session (and Scaffold) implementations is, to be honest. The documentation of TransferMonitoredTrainingSession is just copied from the tf.train.MonitoredTrainingSession. Some of the features implemented there ("checkpoint_filename_with_path") are not even used.

Reply:
> I only wanted to change the behaviour in case the "load_model" is given and nothing else, so I only changed (not load_model_dir) to load_model_dir.

Sure. Thank you for catching our bug in this finetuning implementation.

> I assume that if there is a checkpoint in the specified folder (tf.train.latest_checkpoint(checkpoint_dir)) then we should be in the"continue_learning" mode, and hence the "TransferMonitoredTrainingSession" instead of the tf.train.MonitoredTrainingSession. 

`"continue_learning"` mode should restore whole graph (including current step, learning rates, etc.), so it is implemented in default TensorFlow [`MonitoredSession`](https://www.tensorflow.org/api_docs/python/tf/train/MonitoredSession). 

> Though it's unclear to me what the difference between these Session (and Scaffold) implementations is, to be honest. The documentation of TransferMonitoredTrainingSession is just copied from the tf.train.MonitoredTrainingSession. Some of the features implemented there ("checkpoint_filename_with_path") are not even used.

Yes, this part definitely needs improvement. The most important class here is `TransferSessionManager` which restores trainable variables.

Thank you for the PR!